### PR TITLE
THRIFT-4516: Fix "go vet" warnings for Go 1.10

### PR DESCRIPTION
**Pull request by dcelasun**
Opening this here because my free Travis account has its builds killed after 50 minutes.

### CODE REVIEW

File: lib/go/test/dontexportrwtest/compile_test.go [hunk 1]
```diff
@@ -29,10 +28,10 @@ import (
 func TestReadWriteMethodsArePrivate(t *testing.T) {
 	// This will only compile if read/write methods exist
 	s := NewTestStruct()
-	fmt.Sprintf("%v", s.read)
-	fmt.Sprintf("%v", s.write)
+	_ = s.read
```
Comment (jeking3) [inline lib/go/test/dontexportrwtest/compile_test.go]: Nice.. I had done everything else but I had no idea how to fix this in golang.  Thanks!

**Comment by jeking3** [comment]
I'll keep an eye on this.  Right now the docker images on docker.io have to be generated manually every time they are updated, otherwise the build jobs take more time than they should.

File: lib/go/test/dontexportrwtest/compile_test.go (hunk 1 above)
Comment (dcelasun) [reply lib/go/test/dontexportrwtest/compile_test.go]: The problem is `s.read` and `s.write` are functions and there just isn't any way to print functions as strings, so I had to find another way to use them without trying to stringify them.

**Comment by dcelasun** [comment]
Failures seem unrelated. Should I merge or do you need to something on docker.io?

**Comment by jeking3** [comment]
Go ahead and merge.  After you merge, I update my fork and kick a build on my account to refresh the images.  It's an optimization that saves ~10 minutes per build job.

**Pull request merged**